src/visip/dev/extract_signature.py

```diff
@@ -25,30 +25,34 @@ def unwrap_type(type_hint):
         return None   #raise TypeError("Type annotation is required.")
     #elif type_hint is type(None):
     #    return None
-    elif ti.is_any(type_hint):
+    elif isinstance(type_hint, dtype.DTypeBase):
         return type_hint
-    elif ti.is_base_type(type_hint):
-        return type_hint
-    elif ti.is_dataclass(type_hint):
-        return type_hint
-    elif ti.is_newtype(type_hint):
-        return type_hint
-    else:
-        args = ti.get_args(type_hint)
+    elif isinstance(type_hint, dtype.DTypeGeneric):
+        args = type_hint.get_args()
         if args:
             # Is a generic, process its parameters recursively.
             uargs = tuple(unwrap_type(arg) for arg in args)
-            typing_origin = ti.get_typing_origin(type_hint)
-            return typing_origin[uargs]
+            typing_origin = type(type_hint)
```

Comment:
Document which type_hints are dtype and which typing.
Rather use 'dtype.type' and 'dtype.typing_origin'.